### PR TITLE
A bunch of useful fixes

### PR DESCRIPTION
**Pull request by igor-eremeev**
Hey guys! Got some fixes, maybe you will find some of them worth merging.
1. Add an override for zip alignment bug.
It's and old bug with python being unable to read non-conforming zip archives, http://bugs.python.org/issue14315. Gradle seems to always generate such a file for apk, so it's impossible to use redex on systems with old python. Updating it may be difficult on old systems like Ubuntu 14.04 LTS, for example on CI build agents and so on. This fix allows to overcome this error on systems with bad python, while doing nothing on systems with the good one.
2. Align uncompressed so files for 4098 bytes.
Redexing breaks 4k alignment for uncompressed native libs (https://stackoverflow.com/a/44704840/6109012) because of the repackaging. Add the -p switch for zipalign that does exactly that - creates 4k alignment for .so's with "store" compression.
3. Fix hardcoded dex files limit to enable >100 dex files apk processing.
This was quite a critical bug: redex skips dex files with id grater then 100 when packaging apk. I.e. if you had an apk file with 120 dexes, then redex will produce a file only with first 100 of them, resulting in broken application.

### CODE REVIEW

**Comment by justinjhendrick** [comment]
Thanks! Looks good overall:

1. Sounds like a useful fix.
2. We're going to have to test this one internally and see if it changes things before we could add `-p`. Maybe put it behind a flag to redex? That way we can more easily measure the difference on apk size.
3. That's much better. Thanks for the fix. We've never hit it though because even our largest apps have much less than 100 dexes. What are you doing that creates so many dex files? Is each dex file actually filled to the maximum?

**Comment by igor-eremeev** [comment]
2. Done.
3. This is a result of preDexLibraries (https://goo.gl/9qFLkV). It gets enabled by default if you set minifyEnabled to false, and results in a separate dex file being generated for every aar/jar dependency. Of course we won't ship such a build to production, the main purpose of this is to speed up build. So this only reproduces when you disable minifyEnabled, but enable redex (for example, for debugging build system integration). So, a corner case, really.

**Comment by justinjhendrick** [comment]
Looks good to me. I'm importing internally for review. It should land in master some time today or early next week (with you listed as the author still)

File: redex.py [hunk 1]
```diff
@@ -34,6 +34,17 @@
 from pyredex.utils import abs_glob, make_temp_dir, remove_temp_dirs
 from pyredex.logger import log
 
+# See http://bugs.python.org/issue14315
+def patch_zip_file():
+    old_decode_extra = zipfile.ZipInfo._decodeExtra
+    def decodeExtra(self):
+        try:
+            old_decode_extra(self)
+        except struct.error:
```
Comment (justinjhendrick) [inline redex.py]: Our linter is unhappy about this line. It says `undefined name 'struct'`. Does this work on python2 and python3? I haven't had a chance to run this on my machine yet.

Comment (justinjhendrick) [reply redex.py]: Just a missing `import struct`. I fixed it

**Pull request closed without merging**